protos/idl/service.proto

```diff
@@ -98,3 +110,74 @@ message Tag {
 message Metadata {
     map<string, string> key_map = 1; // key map is a dictionary of key/val strings that represent metadata
 }
+
+// Filter expression that is composed of a combination of single filters
+message FilterExpression {
+    repeated SinglePropertyFilter filters = 1;
+}
+
+// A single property to filter on.
+message SinglePropertyFilter {
+    oneof property_filter {
```

Comment:
Would it be confusing for callers that only some filters actually do anything for each request type?  E.g. passing a DatasetPropertyFilter to ListArtifacts won't do anything - should this be encoded somehow, or do you not see that as a problem?

Reply:
That's a good question. I think it is OK to leave the responsibility of making a sensible filter expression up to the client. I say this because the client can equally do something weird like `artifact_id='x' and artifact_id='y'`. Having said that though, we will definitely do some validation on invalid filter properties like the case you mentioned.

Reply:
Makes sense.  I am a fan of the simplicity / flexibility of the current version.

Reply:
👍 Great thanks!